src/webhook/mutation/pod_mutator/init_container.go

```diff
@@ -27,19 +28,41 @@ func createInstallInitContainerBase(webhookImage, clusterID string, pod *corev1.
 			{Name: config.K8sNamespaceEnv, ValueFrom: kubeobjects.NewEnvVarSourceForField("metadata.namespace")},
 			{Name: config.K8sNodeNameEnv, ValueFrom: kubeobjects.NewEnvVarSourceForField("spec.nodeName")},
 		},
-		SecurityContext: copyUserContainerSecurityContext(pod),
+		SecurityContext: securityContextForInitContainer(pod),
 		Resources:       *dynakube.InitResources(),
 	}
 }
 
-func copyUserContainerSecurityContext(pod *corev1.Pod) *corev1.SecurityContext {
-	var securityContext *corev1.SecurityContext
-	if len(pod.Spec.Containers) == 0 {
-		return securityContext
+func securityContextForInitContainer(pod *corev1.Pod) *corev1.SecurityContext {
+	var securityContext = &corev1.SecurityContext{
+		RunAsNonRoot:             address.Of(true),
+		ReadOnlyRootFilesystem:   address.Of(true),
+		AllowPrivilegeEscalation: address.Of(false),
+		Privileged:               address.Of(false),
+		Capabilities: &corev1.Capabilities{
+			Drop: []corev1.Capability{
+				"ALL",
+			},
+		},
+		SeccompProfile: &corev1.SeccompProfile{
+			Type: corev1.SeccompProfileTypeRuntimeDefault,
+		},
 	}
-	if pod.Spec.Containers[0].SecurityContext != nil {
-		securityContext = pod.Spec.Containers[0].SecurityContext.DeepCopy()
+
+	var containerSecurityContext = pod.Spec.Containers[0].SecurityContext
+	var podSecurityContext = pod.Spec.SecurityContext
+
+	if containerSecurityContext != nil && containerSecurityContext.RunAsUser != nil && containerSecurityContext.RunAsGroup != nil {
+		securityContext.RunAsGroup = containerSecurityContext.RunAsGroup
+		securityContext.RunAsUser = containerSecurityContext.RunAsUser
+	} else if podSecurityContext != nil && podSecurityContext.RunAsUser != nil && podSecurityContext.RunAsGroup != nil {
+		securityContext.RunAsGroup = containerSecurityContext.RunAsGroup
+		securityContext.RunAsUser = containerSecurityContext.RunAsUser
+	} else {
+		securityContext.RunAsGroup = address.Of(int64(1001))
+		securityContext.RunAsUser = address.Of(int64(1001))
```

Comment:
could we abstract it into a separate function ?

abstracting the ifs might be enough, so to have something `if hasUserAndGroupSet(containerSecurityContext) ...`